### PR TITLE
restructuring inspector-driver integration

### PR DESCRIPTION
**Pull request by deven96**

### CODE REVIEW

**Comment by deven96** [comment]
Oh yeah I was able to run the simple dashboard branch on Imani's windows 11 cmdprompt and it worked @MeNsaaH

File: driver/driver.go [hunk 1]
```diff
@@ -1,18 +1,32 @@
...
+	IsLinux   bool
+	IsDarwin  bool
+	IsWeb     bool
+	Name      string
+	Extra     string
+}
+
 type fields struct {
 	// Supported inspector representations for specific driver
 	Supported []string
```
Comment (MeNsaaH) [inline driver/driver.go:15]: We should be able to take this out, do we still need it?

File: driver/driver.go [hunk 2]
```diff
@@ -1,18 +1,32 @@
...
 	// Selected inspector representations
 	Selected []string
```
Comment (MeNsaaH) [inline driver/driver.go:17]: And this as well

File: driver/driver.go [hunk 3]
```diff
@@ -1,18 +1,32 @@
...
+type SystemDetails struct {
+	IsWindows bool
+	IsLinux   bool
+	IsDarwin  bool
+	IsWeb     bool
+	Name      string
+	Extra     string
+}
+
 type fields struct {
```
Comment (MeNsaaH) [inline driver/driver.go:13]: you think we can make this something else or give it a more concrete name? like `driverBase`

File: driver/ssh.go [hunk 4]
```diff
@@ -27,7 +27,8 @@ type SSH struct {
 	// Check known hosts (only disable for tests
 	CheckKnownHosts bool
 	// set environmental vars for server e.g []string{"DEBUG=1", "FAKE=echo"}
-	Vars []string
+	Vars          []string
```
Comment (MeNsaaH) [inline driver/ssh.go:30]: Should be `EnvVars`. `Vars` is a bit ambigous.

File: driver/ssh.go [hunk 5]
```diff
@@ -36,33 +37,37 @@ func (d *SSH) String() string {
...
-	}
-	if d.CheckKnownHosts {
-		callback, err = goph.DefaultKnownHosts()
+	if d.SessionClient == nil {
+		var err error
+		var client *goph.Client
+		var callback ssh.HostKeyCallback
+		if d.Port != 0 {
+			port = d.Port
+		}
```
Comment (MeNsaaH) [inline driver/ssh.go:46]: Default value if not specified is 22. Is this used if it's not specified?

File: driver/ssh.go [hunk 6]
```diff
@@ -36,33 +37,37 @@ func (d *SSH) String() string {
...
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			callback = ssh.InsecureIgnoreHostKey()
+		}
+		client, err = goph.NewConn(&goph.Config{
+			User:     d.User,
+			Addr:     d.Host,
+			Port:     uint(port),
```
Comment (MeNsaaH) [inline driver/ssh.go:62]: We should set the default for port to 22 above

File: driver/ssh.go [hunk 7]
```diff
@@ -72,7 +77,7 @@ func (d *SSH) ReadFile(path string) (string, error) {
 }
 
 func (d *SSH) RunCommand(command string) (string, error) {
-	// FIXME: Do we retain client across all command runs?
+	// FIXME: provide refresh interface to somehow refresh d.Client if d.SessionClient somehow gets dropped
```
Comment (MeNsaaH) [inline driver/ssh.go:80]: I think we sould retain the client across all commands by the driver. That way reduce the number of connections we have to make. We should probably not call `defer client.Close()` and close all the connections when we're done

File: driver/ssh.go [hunk 8]
```diff
@@ -91,20 +96,29 @@ func (d *SSH) RunCommand(command string) (string, error) {
...
-		KeyPass:         "",
-		CheckKnownHosts: false,
-		fields: fields{
-			PollInterval: 5,
-		},
+func (d *SSH) GetDetails() SystemDetails {
+	if d.Info == nil {
+		uname, err := d.RunCommand(`uname`)
+		// try windows command
+		if err != nil {
```
Comment (MeNsaaH) [inline driver/ssh.go:103]: Probably need to check the error could be other errors. Like ssh connection errors.

File: driver/web.go [hunk 9]
```diff
@@ -21,7 +21,7 @@ const (
 	GET Request = "GET"
 )
 
-// Web : Driver for handling ssh executions
+// Web : Driver for handling web executions
```
Comment (MeNsaaH) [inline driver/web.go:24]: Why exactly do we have web drivers again 🙈 ?

**Review by MeNsaaH — CHANGES_REQUESTED** [review]

File: driver/ssh.go (hunk 5 above)
Comment (deven96) [reply driver/ssh.go:46]: It is already specified as var on the entire file

File: driver/ssh.go (hunk 7 above)
Comment (deven96) [reply driver/ssh.go:80]: When exactly do we close the connections

File: driver/ssh.go (hunk 8 above)
Comment (deven96) [reply driver/ssh.go:103]: And then what do we do at that point

File: driver/web.go (hunk 9 above)
Comment (deven96) [reply driver/web.go:24]: Lol... to help monitor web applications (we haven't fleshed out the inspectors for those but it is very valid)

**Pull request merged**